examples/data-solutions/dp-foundation/05-dtl-main.tf

```diff
@@ -0,0 +1,178 @@
+# limitations under the License.
+
+locals {
+  group_iam_dtl = {
+    "${local.groups.data-engineers}" = [
+      "roles/bigquery.dataEditor",
+      "roles/storage.admin",
+    ],
+    "${local.groups.data-analysts}" = [
+      "roles/bigquery.dataViewer",
+      "roles/bigquery.jobUser",
+      "roles/bigquery.user",
+      "roles/datacatalog.viewer",
+      "roles/datacatalog.tagTemplateViewer",
+      "roles/storage.objectViewer",
+    ]
+  }
+  iam_dtl = {
+    "roles/bigquery.dataEditor" = [
+      module.lod-sa-df-0.iam_email,
+      module.trf-sa-df-0.iam_email,
+      module.trf-sa-bq-0.iam_email,
+      module.orc-sa-cmp-0.iam_email,
+    ]
+    "roles/bigquery.jobUser" = [
+      module.lod-sa-df-0.iam_email,
+      module.trf-sa-df-0.iam_email,
+    ]
+    "roles/storage.admin" = [
+      module.lod-sa-df-0.iam_email,
+      module.trf-sa-df-0.iam_email,
+    ]
+    "roles/storage.objectCreator" = [
+      module.lod-sa-df-0.iam_email,
+      module.trf-sa-df-0.iam_email,
+      module.trf-sa-bq-0.iam_email,
+      module.orc-sa-cmp-0.iam_email,
+    ]
+    "roles/storage.objectViewer" = [
+      module.trf-sa-df-0.iam_email,
+      module.trf-sa-bq-0.iam_email,
+      module.orc-sa-cmp-0.iam_email,
+    ]
+  }
+  prefix_dtl = "${var.prefix}-dtl"
+}
+
+###############################################################################
+#                                 Project                                     #
+###############################################################################
+
+module "dtl-0-prj" {
+  source          = "../../../modules/project"
+  name            = var.project_create == null ? var.project_id["datalake"] : "${var.project_id["datalake"]}-0"
+  parent          = try(var.project_create.parent, null)
+  billing_account = try(var.project_create.billing_account_id, null)
+  project_create  = var.project_create != null
+  prefix          = var.project_create == null ? null : var.prefix
+  # additive IAM bindings avoid disrupting bindings in existing project
+  iam          = var.project_create != null ? local.iam_dtl : {}
+  iam_additive = var.project_create == null ? local.iam_dtl : {}
+  group_iam    = local.group_iam_dtl
+  services = concat(var.project_services, [
+    "bigquery.googleapis.com",
+    "bigqueryreservation.googleapis.com",
+    "bigquerystorage.googleapis.com",
+    "cloudkms.googleapis.com",
+    "compute.googleapis.com",
+    "dataflow.googleapis.com",
+    "pubsub.googleapis.com",
+    "servicenetworking.googleapis.com",
+    "storage.googleapis.com",
+    "storage-component.googleapis.com"
+  ])
+  service_encryption_key_ids = {
+    bq      = [try(local.service_encryption_keys.bq, null)]
+    storage = [try(local.service_encryption_keys.storage, null)]
+  }
+}
+
+module "dtl-1-prj" {
+  source          = "../../../modules/project"
+  name            = var.project_create == null ? var.project_id["datalake"] : "${var.project_id["datalake"]}-1"
```

Comment:
What if project create is false?

Reply:
project_create is an object with 2 values. False can't be assigned.